src/xb-builder-source.c

```diff
@@ -298,6 +314,10 @@ xb_builder_source_add_node_func (XbBuilderSource *self,
  * Adds a function that can be used to convert files loaded with
  * xb_builder_source_load_xml().
  *
+ * This function is now deprecated, and xb_builder_source_add_adapter() can be
+ * used instead. The latter allows chaining streams rather than just one #GFile
+ * to #GInputStream conversion.
```

Comment:
Maybe worth doing a g_warning or similar that this is a deprecated function to discourage use

Reply:
I think build time it's useful, but not so much runtime. When you build an old gnome-software with this branch you get:

    ../plugins/flatpak/gs-flatpak.c: In function ‘gs_flatpak_load_desktop_fn’:
    ../plugins/flatpak/gs-flatpak.c:585:2: warning: ‘xb_builder_source_add_converter’ is deprecated: Use 'xb_builder_source_add_adapter' instead [-Wdeprecated-declarations]